pkg/util/util_test.go

```diff
@@ -499,6 +505,19 @@ func (_ validNetworkInterface) Addrs(intf *net.Interface) ([]net.Addr, error) {
 	return ifat, nil
 }
 
+type validNetworkInterfaceWithoutLinkLocal struct {
```

Comment:
Maybe. It checks that it *doesn't* select link local, but the test really is about if the network interface is selected properly with a link-local in the pool...

e2e is broken though so I think that's the bigger issue.